### PR TITLE
New Registration Behaviour

### PR DESCRIPTION
**Pull request by ac-jorellanaf**
Registration page now forcefully appears if the logged-in user is not registered to GHGA

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,12 +14,55 @@ import Login from "./components/login/login";
...
-      <Routes>
+
+function RouterWrapper() {
+  
+  let location = useLocation();
+  const navigate = useNavigate()
+
+  useLayoutEffect(() => {
+    authService.getUser().then((user) => {
+      if ((!user?.id || user.changed) && location.pathname !== "/register" && user !== null) {
```
Comment (Cito) [inline src/App.tsx:28]:
```suggestion
      if (user && (!user.id || user.changed) && location.pathname !== "/register") {
```

First check whether user exists before checking user properties.

File: src/App.tsx [hunk 2]
```diff
@@ -14,12 +14,55 @@ import Login from "./components/login/login";
...
+  //   await authService.logout();
+  //   const lastUrl = sessionStorage.getItem("lastUrl");
+  //   lastUrl ? window.location.href = lastUrl : navigate("/");
+  // };
+
+  // window.onunload = () => {
+  //   if (location.pathname === "/register") {
+  //     logout();
+  //   }
+  // };
```
Comment (Cito) [inline src/App.tsx:62]:
```suggestion
```

Clean up unnecessary code.

File: src/components/header/headerNavbar.tsx [hunk 3]
```diff
@@ -143,152 +112,7 @@ const HeaderNavbar = () => {
...
-              <Overlay target={target.current} show={show} placement="left">
-                {(props) => (
-                  <Tooltip id="overlay-example" {...props}>
-                    Log In
-                  </Tooltip>
-                )}
-              </Overlay>
-            </>
-          )}
+          <LoginButton/>
```
Comment (Cito) [inline src/components/header/headerNavbar.tsx:115]: Good idea to move this to a separate component.

File: src/components/header/loginButton.tsx [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+  const logout = async () => {
+    await authService.logout();
+    setUser(null);
+    window.location.reload();
+  };
+
+  return (
+    <div>
+      {user ? (
+        !user?.id || user.changed ? (
```
Comment (Cito) [inline src/components/header/loginButton.tsx:48]:
```suggestion
        !user.id || user.changed ? (
```

File: src/components/header/loginButton.tsx [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+              show={showPopover}
+              overlay={
+                <Popover
+                  id={"ProfilePopover"}
+                  className="border border-2 border-tertiary"
+                >
+                  <Popover.Body className="text-center fs-6 px-4">
+                    <p>
+                      You need to complete your {user.changed ? "re-" : ""}
+                      registration to the GHGA Data Portal before you can start
```
Comment (Cito) [inline src/components/header/loginButton.tsx:64]:
```suggestion
                      registration with the GHGA Data Portal before you can start
```

File: src/components/header/loginButton.tsx [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+                    </p>
+                    <p className="mt-3">
+                      <a
+                        href="https://profile.aai.lifescience-ri.eu/profile"
+                        target="_blank"
+                        rel="noreferrer"
+                      >
+                        Manage your
+                        <br />
+                        LS Login Account
```
Comment (Cito) [inline src/components/header/loginButton.tsx:162]:
```suggestion
                        LS Login account
```

File: src/services/auth.ts [hunk 7]
```diff
@@ -89,7 +89,7 @@ class AuthService {
    */
   private notify(user: User | null): void {
     document.dispatchEvent(new CustomEvent("auth", { detail: user }));
-    console.log("New user object:", user);
+    //console.log("New user object:", user);
```
Comment (Cito) [inline src/services/auth.ts:92]:
```suggestion
```

Clean up.

**Review by Cito — DISMISSED** [review]
Just some minor suggestions.

**Review by dontseyit — APPROVED** [review]
I approved, but while testing on my local I found the following, it might be expected behaviour or the reason might not be related to this code. 
First I logged in with a registered email/account. Then I went to profile settings on aai.lifescience-ri.eu and updated my email address (to my personal address). I was still able to browse and use the Data Portal, until when I logged out and logged in back, then I was redirected to the registration page.

**Review by Cito — APPROVED** [review]

**Pull request merged**